auth/auth.go

```diff
@@ -36,6 +36,14 @@ import (
 	authorizationclient "k8s.io/client-go/kubernetes/typed/authorization/v1beta1"
 )
 
+type AuthInterface interface {
```

Comment:
nit `AuthInterface interface` stutters, can we call this `Authorizer`?

Reply:
It encapsulates both authentication and authorization, so `Authorizer` would not give the right notion, I think.

Reply:
I agree with Frederic. Like anything there can be exceptions to these naming guidelines. One thing is I always find it funny when types have the type in the name. We could call this simply `Auther`.

Reply:
If it encapsulates two separate concerns, it smells like the interface is too big.

Just asking: Do we need to expose this interface at all? Why not simply returning `(*kubeRBACProxyAuth, error)`? The interface declaration could be anonymous, ensuring compile time safety, the caller could then type cast to `Authorizer` and/or `Authenticator` at will.

All of this is type bike shedding though, ignore it at will ;-)

Reply:
I like `Auther`, let's go with that.

Reply:
@s-urbaniak agreed. I removed it.